analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.EF6.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public async Task DatabaseMethods(Database database, string query, int x)
+        {
+            database.SqlQuery<Program>(query + x); // Noncompliant {{Make sure using a dynamically formatted SQL query is safe here.}}
+            database.SqlQuery(typeof(Program), query + x);                                             // Noncompliant
+            database.SqlQuery<Program>($"{query} {x}");                                                // Noncompliant
+            database.SqlQuery<Program>(string.Format("Select * from Program Where Id={1}", x));        // Noncompliant
+
+            database.ExecuteSqlCommand(query + x);                                                     // Noncompliant
+            database.ExecuteSqlCommand(TransactionalBehavior.EnsureTransaction, query + x);            // Noncompliant
+            await database.ExecuteSqlCommandAsync(query + x);                                          // Noncompliant
+            await database.ExecuteSqlCommandAsync(TransactionalBehavior.EnsureTransaction, query + x); // Noncompliant
+        }
+
+        public void DbSetMethods(DbSet set, string query, int x)
+        {
+            set.SqlQuery(query + x);    // Noncompliant {{Make sure using a dynamically formatted SQL query is safe here.}}
+            set.SqlQuery(query + x, x); // Noncompliant
```

Comment:
This is unclear to me if it triggers because of `query + x` or the second argument `x`. Let's split the arguments. I assume `set.SqlQuery("constant", x)` is compliant (?)